xsl/xccdf-resources/openscap.js

```diff
@@ -108,12 +108,131 @@ function ruleSearch()
         $("#search-matches").html("No rules match your search criteria!");
 }
 
+var is_original = true;
+var original_treetable = null;
+
 $(document).ready( function() {
     $("#result-details").hide();
     $(".js-only").show();
     $(".toggle-rule-display").each(function(){
         toggleRuleDisplay(this);
     });
-
-    $(".treetable").treetable({ column: 0, expandable: true, initialState : "expanded", indent : 0 });
+    original_treetable = $(".treetable").clone();
+    $(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+    is_original = true;
 });
+
+function Reset() {
+	if (!is_original) {
+		$(".treetable").remove();
+		$("#rule-overview").append(original_treetable.clone());
+		$(".treetable").treetable({ column: 0, expandable: true, clickableNodeNames: true, initialState : "expanded", indent : 0 });
+		$(".toggle-rule-display").each(function(){
+			toggleRuleDisplay(this);
+		});
+		is_original = true;
+	}
+}
+
+function NewGroupLine(group_name)
+{
+	return "<tr class=\"rule-overview-inner-node\" data-tt-id=\"" + group_name + "\">" +
+		"<td colspan=\"3\"><strong>" + group_name + "</strong></td></tr>";
+}
+
+var KeysEnum = {
+	DEFAULT: "default",
+	SEVERITY : "severity",
+	RESULT: "result",
+	NIST: "data-nist-id",
+	DISA: "data-disa-id"
+};
+
+function GetTargetGroupsList(rule, key)
+{
+	/* This function returns an array of target groups indentifiers */
+	if (key == KeysEnum.SEVERITY) {
+		var severity = rule.children("td:nth-child(2)").text();
+		return [severity];
+	} else if (key == KeysEnum.RESULT) {
+		var result = rule.children(".rule-result").text();
+		return [result];
+	}
+	else {
+		var ref_list = rule.attr(key);
+		if (!ref_list) {
+			ref_list = "unknown";
+		}
+		return ref_list.split(",");
+	}
+}
+
+function SortGroups(groups, key)
+{
+	if (key == KeysEnum.SEVERITY) {
+		return ["high", "medium", "low"];
+	} else if (key == KeysEnum.DISA) {
+		return groups.sort(function(a, b){return a-b});
+	} else if (key == KeysEnum.NIST) {
+		return groups.sort(function(a, b){
+			var regex = /(\w\w)-(\d+)(.*)/;
+			var a_parts = regex.exec(a);
+			var b_parts = regex.exec(b);
+			if (a_parts == null)
+				return 1;
+			if (b_parts == null)
+				return -1;
+			var result = a_parts[1].localeCompare(b_parts[1]);
+			if (result != 0) {
+				return result;
+			} else {
+				result = a_parts[2] - b_parts[2];
+				if (result != 0) {
+					return result;
+				} else {
+					return a_parts[3].localeCompare(b_parts[3]);
+				}
+			}
+		});
+	} else {
+		return groups;
+	}
+}
+
+function GroupBy(key) {
+	/* We must process grouping upon the original table.
```

Comment:
I think the comment describes whole function. Why is it inside of? http://stackoverflow.com/a/10126365